packages/perseus/src/util/parse-perseus-json/perseus-parsers/expression-widget.ts

```diff
@@ -51,3 +58,58 @@ export const parseExpressionWidget: Parser<ExpressionWidget> = parseWidget(
         buttonsVisible: optional(enumeration("always", "never", "focused")),
     }),
 );
+
+const parseExpressionWidgetV0 = parseWidgetWithVersion(
+    optional(object({major: constant(0), minor: number})),
+    constant("expression"),
+    object({
+        functions: array(string),
+        times: boolean,
+        visibleLabel: optional(string),
+        ariaLabel: optional(string),
+        form: boolean,
+        simplify: boolean,
+        value: string,
+        buttonSets: array(
+            enumeration(
+                "basic",
+                "basic+div",
+                "trig",
+                "prealgebra",
+                "logarithms",
+                "basic relations",
+                "advanced relations",
+            ),
+        ),
+        buttonsVisible: optional(enumeration("always", "never", "focused")),
+    }),
+);
+
+function migrateV0ToLatest(widget: ParsedValue<typeof parseExpressionWidgetV0>, ctx: ParseContext): ParseResult<ExpressionWidget> {
+    const {options} = widget;
+    return ctx.success({
+        ...widget,
+        version: Expression.version,
+        options: {
+            times: options.times,
+            buttonSets: options.buttonSets,
+            functions: options.functions,
+            buttonsVisible: options.buttonsVisible,
+            visibleLabel: options.visibleLabel,
+            ariaLabel: options.ariaLabel,
+
+            answerForms: [
+                {
+                    considered: "correct",
+                    form: options.form,
+                    simplify: options.simplify,
+                    value: options.value,
+                },
+            ],
+        },
+    })
+}
+
+export const parseExpressionWidget: Parser<ExpressionWidget> =
+    union(pipeParsers(parseExpressionWidgetV0).then(migrateV0ToLatest).parser)
```

Comment:
So this will attempt to parse using the v0 and migration parsers and fall back to the `parseExpressionWidgetLatest` if that fails, right? 

This might be extremely over-the-top optimization, but I wonder if we should order these in the opposite order given we'll more likely get a latest version set of options rather than v0 options?  🤔